### PR TITLE
Export cardano-testnet API from Cardano.Testnet

### PR DESCRIPTION
**Pull request by MarcFontaine**
Resolves: #4602

### CODE REVIEW

**Comment by MarcFontaine** [comment]
https://github.com/input-output-hk/cardano-node/issues/4602

File: cardano-testnet/cardano-testnet.cabal [hunk 1]
```diff
@@ -57,27 +57,28 @@ library
...
-                        Parsers.Shelley
-                        Parsers.Version
+  exposed-modules:      Cardano.Testnet
+                        Testnet.Byron
                         Util.Assert
                         Util.Base
                         Util.Ignore
                         Util.Process
                         Util.Runtime
+                        Parsers
```
Comment (Jimbo4350) [inline cardano-testnet/cardano-testnet.cabal:67]: I don't think `Parsers` needs to be exposed

File: cardano-testnet/cardano-testnet.cabal [hunk 2]
```diff
@@ -57,27 +57,28 @@ library
...
   hs-source-dirs:       src
-  exposed-modules:      Parsers
-                        Parsers.Babbage
-                        Parsers.Byron
-                        Parsers.Cardano
-                        Parsers.Shelley
-                        Parsers.Version
+  exposed-modules:      Cardano.Testnet
+                        Testnet.Byron
                         Util.Assert
```
Comment (Jimbo4350) [inline cardano-testnet/cardano-testnet.cabal:62]: Let's change `Util.x` to `Testnet.Util.x`. We will need to rename `Testnet.Util` to something else or put the functions/types in a suitable `Testnet.Util.x` module

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]

**Review by Jimbo4350 — APPROVED** [review]
LGTM! Just squash the commits

File: cardano-testnet/cardano-testnet.cabal (hunk 1 above)
Comment (MarcFontaine) [reply cardano-testnet/cardano-testnet.cabal]: Turned out it was needed for `cabal build cardano-testnet:exe:cardano-testnet`.
In any case, I renamed `Parsers` to `Testnet.Parsers` and kept it in the exported API.

File: cardano-testnet/src/Testnet/Parsers.hs [hunk 3]
```diff
@@ -1,4 +1,8 @@
-module Parsers where
+module Testnet.Parsers
+(
+  commands
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Parsers.hs:3]: This does not follow the typical module export style. Please fix.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]

**Review by Jimbo4350 — APPROVED** [review]
Good work :+1:

File: cardano-testnet/src/Testnet/Parsers.hs [hunk 4]
```diff
@@ -1,4 +1,6 @@
-module Parsers where
+module Testnet.Parsers
```
Comment (Jimbo4350) [inline cardano-testnet/src/Testnet/Parsers.hs:1]: :+1:

File: cardano-testnet/test/Test/Cli/KesPeriodInfo.hs [hunk 5]
```diff
@@ -129,13 +121,13 @@ hprop_kes_period_info = H.integration . H.runFinallies . H.workspace "chairman"
...
   poolownerstakeaddr <- filter (/= '\n')
-                          <$> H.execCli
+                          <$> execCli
                                 [ "stake-address", "build"
                                 , "--stake-verification-key-file", poolownerstakekey
                                 , "--testnet-magic", show @Int testnetMagic
                                 ]
 
-  poolowneraddresswstakecred <- H.execCli [ "address", "build"
+  poolowneraddresswstakecred <- execCli [ "address", "build"
```
Comment (Jimbo4350) [inline cardano-testnet/test/Test/Cli/KesPeriodInfo.hs:130]: List alignment

File: cardano-testnet/test/Test/Cli/KesPeriodInfo.hs [hunk 6]
```diff
@@ -144,20 +136,20 @@ hprop_kes_period_info = H.integration . H.runFinallies . H.workspace "chairman"
   poolcoldSkey <- H.note $ tempAbsPath </> "node-pool1/shelley/operator.skey"
 
   stakePoolId <- filter ( /= '\n') <$>
-                   H.execCli [ "stake-pool", "id"
+                   execCli [ "stake-pool", "id"
```
Comment (Jimbo4350) [inline cardano-testnet/test/Test/Cli/KesPeriodInfo.hs:139]: List alignment

File: cardano-testnet/test/Test/Cli/KesPeriodInfo.hs [hunk 7]
```diff
@@ -224,12 +216,12 @@ hprop_kes_period_info = H.integration . H.runFinallies . H.workspace "chairman"
   UTxO utxoWithStaking1 <- H.noteShowM $ H.jsonErrorFail $ J.fromJSON @(UTxO AlonzoEra) utxoWithStaking1Json
   txinForStakeReg <- H.noteShow =<< H.headM (Map.keys utxoWithStaking1)
 
-  void $ H.execCli [ "stake-address", "registration-certificate"
+  void $ execCli [ "stake-address", "registration-certificate"
```
Comment (Jimbo4350) [inline cardano-testnet/test/Test/Cli/KesPeriodInfo.hs:219]: List alignment

File: cardano-testnet/test/Test/Cli/KesPeriodInfo.hs [hunk 8]
```diff
@@ -241,7 +233,7 @@ hprop_kes_period_info = H.integration . H.runFinallies . H.workspace "chairman"
     , "--out-file", work </> "key-registration-cert.txbody"
     ]
 
-  void $ H.execCli [ "transaction", "sign"
+  void $ execCli [ "transaction", "sign"
```
Comment (Jimbo4350) [inline cardano-testnet/test/Test/Cli/KesPeriodInfo.hs:236]: List alignment

**Pull request merged**